pkg/controllers/allocation/controller.go

```diff
@@ -92,13 +92,23 @@ func (c *Controller) Reconcile(ctx context.Context, object client.Object) (recon
 		packings = append(packings, c.packer.Pack(ctx, constraintGroup, instanceTypes)...)
 	}
 
-	// 4. Create packedNodes for packings
-	packedNodes, err := c.cloudProvider.Create(ctx, provisioner, packings)
+	// 4. Complement provisioner with (dynamic) default values, which must not
+	//    be hoisted into the original CRD as they might change with each reconciliation
+	//    loop iteration.
+	provisionerWithDefaults, err := provisioner.WithDefaults()
```

Comment:
It strikes me that it might be useful to include cluster.KubernetesVersion in this mechanism

Reply:
Thoughts on running this as the very first step?

Reply:
Also, consider calling this WithDynamicDefaults to differentiate between SetDefaults (which are static).

Reply:
> It strikes me that it might be useful to include cluster.KubernetesVersion in this mechanism

What do we need it for and where would it be stored (in which struct)?

Reply:
> Thoughts on running this as the very first step?

I am not sure what you have in mind, can your please elaborate a bit more.

Reply:
You understood perfectly :)